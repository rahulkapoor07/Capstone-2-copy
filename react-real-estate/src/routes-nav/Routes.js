import React from "react";
import {Route, Switch, Redirect} from "react-router-dom";
import HomePage from "../homePage/HomePage";
import LoginForm from "../auth/LoginForm";
import RegisterForm from "../auth/RegisterForm";
import BuyList from "../buy/BuyList";
import SoldList from "../Sold/SoldList";
import UserProfile from "../userProfile/UserProfile";
import RentList from "../Rent/RentList";
import AgentsList from "../agents/AgentsList";
import MortgageCalc from "../mortgage/MortgageCalc";
import HomeDetail from "../userProfile/HomeDetail";
import EditProfile from "../editProfile/EditProfile";
import AgentDetails from "../agents/AgentDetails";


const Routes = ({login, signUp, setHomeModalOpen, setIsMortgageOpen,mortgageCalcFunc,setToken,
    setMortgageData})=>{
    return (
        <Switch>
        <Route exact path="/">
            <HomePage />
        </Route>

        <Route exact path="/login">
            <LoginForm login={login} setHomeModalOpen = {setHomeModalOpen}/>
        </Route>

        <Route exact path="/register">
            <RegisterForm signUp={signUp} setHomeModalOpen={setHomeModalOpen}/>
        </Route>

        <Route exact path="/buy">
            <BuyList/>
        </Route>

        <Route exact path="/sold">
            <SoldList/>
        </Route>

        <Route exact path="/agents">
            <AgentsList/>
        </Route>

        <Route exact path="/mortgage-calculator">
            <MortgageCalc setIsMortgageOpen={setIsMortgageOpen} mortgageCalcFunc={mortgageCalcFunc}
            setMortgageData={setMortgageData}/>
        </Route>

        <Route exact path="/rent">
            <RentList/>
        </Route>

        <Route exact path="/profile/:username">
            <UserProfile />
        </Route>

        <Route exact path="/profile/:username/:property_id">
            <HomeDetail />
        </Route>

        <Route exact path="/edit-profile">
            <EditProfile setToken={setToken}/>
        </Route>

        <Route exact path="/agents/:agent_id">
            <AgentDetails />
        </Route>

        <Redirect to="/" />
        </Switch>
    )
}

export default Routes;