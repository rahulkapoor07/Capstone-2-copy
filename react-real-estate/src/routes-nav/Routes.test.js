import { render } from '@testing-library/react';
import Routes from './Routes';
import {MemoryRouter} from "react-router-dom";


test("render without crashing", ()=>{
    render(<MemoryRouter><Routes/></MemoryRouter>);
});

test("render without crashing", ()=>{
    const {asFragment} = render(<MemoryRouter><Routes/></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
});