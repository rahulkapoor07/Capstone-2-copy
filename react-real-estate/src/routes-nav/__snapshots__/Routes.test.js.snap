// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render without crashing 1`] = `
<DocumentFragment>
  <div>
    <div
      class="page container text-center"
    >
      <h1
        class="mb-4 font-weight-bold"
      >
        US Real Estate
      </h1>
      <p
        class="lead"
      >
        Find your dream place to live!
      </p>
      <p>
        <a
          class="btn btn-primary font-weight-bold mr-3"
          href="/login"
        >
          Log in
        </a>
        <a
          class="btn btn-primary font-weight-bold"
          href="/register"
        >
          Sign up
        </a>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
